File: android/app/src/main/java/com/example/chaircharge/data/Charger.kt
package com.example.chaircharge.data

import com.google.gson.annotations.JsonAdapter

data class Charger(
    val id: String?,
    val name: String,
    val address: String,
    @field:JsonAdapter(FlexibleDoubleAdapter::class)
    val lat: Double?,
    @field:JsonAdapter(FlexibleDoubleAdapter::class)
    val lng: Double?,
    val install_place: String?,
    val install_type: String?,
    @field:JsonAdapter(FlexibleBooleanAdapter::class)
    val is_indoor: Boolean?,
    @field:JsonAdapter(FlexibleBooleanAdapter::class)
    val is_outdoor: Boolean?,
    @field:JsonAdapter(FlexibleBooleanAdapter::class)
    val is_movable: Boolean?,
    val contact_phone: String?,
    val distance_m: Double? = null
)

File: backend/app/models/charger.py
from typing import Optional

from pydantic import BaseModel


class Charger(BaseModel):
    id: str
    name: str
    address: str
    lat: float
    lng: float
    install_place: Optional[str] = None
    install_type: Optional[str] = None
    is_indoor: bool = False
    is_outdoor: bool = False
    is_movable: bool = False
    weekday_open: Optional[str] = None
    weekday_close: Optional[str] = None
    saturday_open: Optional[str] = None
    saturday_close: Optional[str] = None
    holiday_open: Optional[str] = None
    holiday_close: Optional[str] = None
    is_24h: bool = False
    simultaneous_count: Optional[int] = None
    air_pump_available: bool = False
    phone_charge_available: bool = False
    contact_phone: Optional[str] = None
    data_date: Optional[str] = None
    review_status: Optional[str] = None
    review_notes: Optional[str] = None


class ChargerWithDistance(Charger):
    distance_m: float
